pkg/base1/cockpit.js

```diff
@@ -1047,6 +1047,26 @@ function basic_scope(cockpit, jquery) {
      *    immediately. Wait until control is returned to the browser.
      */
 
+    var call_queue = [];
+    var call_timeout = null;
+
+    function call_drain() {
+        call_timeout = null;
+        var func;
+        for (;;) {
```

Comment:
> None of the callers do this. The callers are below. But since you brought it up, I've made the code more verbose to make it clear that this is not a general purpose facility.

Just rereading @larsu's comments. Looks like this will need to be a general purpose facility to fix that issue.